python/mxnet/profiler.py

```diff
@@ -54,5 +60,280 @@ def profiler_set_state(state='stop'):
+        """
+        self.name = name
+        self.handle = ProfileHandle()
+        check_call(_LIB.MXProfileCreateDomain(c_str(self.name), ctypes.byref(self.handle)))
+
+    def __str__(self):
+        return self.name
+
+    def new_task(self, name):
+        """Create new Task object owned by this domain
+            Parameters
+            ----------
+            name : string
+                Name of the task
+        """
+        return Task(self, name)
+
+    def new_frame(self, name):
+        """Create new Frame object owned by this domain
+            Parameters
+            ----------
+            name : string
+                Name of the frame
+        """
+        return Frame(self, name)
+
+    def new_counter(self, name, value=None):
+        """Create new Counter object owned by this domain
+            Parameters
+            ----------
+            name : string
+                Name of the counter
+        """
+        return Counter(self, name, value)
+
+    def new_marker(self, name):
+        """Create new Marker object owned by this domain
+            Parameters
+            ----------
+            name : string
+                Name of the marker
+        """
+        return Marker(self, name)
+
+class Task(object):
+    """Profiling Task class
+    A task is a logical unit of work performed by a particular thread.
+    Tasks can nest; thus, tasks typically correspond to functions, scopes, or a case block
+    in a switch statement.
+    You can use the Task API to assign tasks to threads.
+    """
+    def __init__(self, domain, name):
+        """Profiling Task class constructor.
+            Parameters
+            ----------
+            domain : Domain object
+                Domain to which this object belongs
+            name : string
+                Name of the task
+        """
+        self.name = name
+        self.handle = ProfileHandle()
+        check_call(_LIB.MXProfileCreateTask(domain.handle,
+                                            c_str(self.name),
+                                            ctypes.byref(self.handle)))
+
+    def __del__(self):
+        if self.handle is not None:
+            check_call(_LIB.MXProfileDestroyHandle(self.handle))
+
+    def start(self):
+        """Start timing scope for this object"""
+        check_call(_LIB.MXProfileDurationStart(self.handle))
+
+    def stop(self):
+        """Stop timing scope for this object"""
+        check_call(_LIB.MXProfileDurationStop(self.handle))
+
+    def __str__(self):
+        return self.name
+
+
+class Frame(object):
+    """Profiling Frame class
+    Use the frame API to insert calls to the desired places in your code and analyze
+    performance per frame, where frame is the time period between frame begin and end points.
+    When frames are displayed in Intel VTune Amplifier, they are displayed in a
+    separate track, so they provide a way to visually separate this data from normal task data.
```

Comment:
add a note explaining the differences with task

Reply:
ok

Reply:
done